tests/tests_core/test_statistical_test.py

```diff
@@ -0,0 +1,70 @@
+import unittest
+import numpy as np
+
+from expan.core.statistical_test import *
+from expan.core.util import generate_random_data
+
+class StatisticalTestCase(unittest.TestCase):
+    def setUp(self):
+        np.random.seed(41)
+        self.data, self.metadata = generate_random_data()
+
+    def tearDown(self):
+        pass
+
+    def test_setup_single_test(self):
+        variants = Variants('variant', 'control', 'treatment')
+        mobile = FeatureFilter('device_type', 'mobile')
+        kpi = KPI('revenue')
+
+        test_revenue_overall = StatisticalTest(kpi, [], variants)
+        test_revenue_mobile = StatisticalTest(kpi, [mobile], variants)
+
+        self.assertEqual(test_revenue_overall.kpi.name, "revenue")
+        self.assertEqual(test_revenue_overall.variants.control_name, "control")
+        self.assertEqual(test_revenue_overall.variants.treatment_name, "treatment")
+
+        self.assertEqual(test_revenue_mobile.kpi.name, "revenue")
+        self.assertEqual(test_revenue_mobile.variants.control_name, "control")
+        self.assertEqual(test_revenue_mobile.variants.treatment_name, "treatment")
+        self.assertEqual(test_revenue_mobile.features[0].column_name, "device_type")
+        self.assertEqual(test_revenue_mobile.features[0].column_value, "mobile")
+
+    def test_setup_multiple_test_suite(self):
+        variants = Variants('variant', 'control', 'treatment')
+        mobile = FeatureFilter('device_type', 'mobile')
+        desktop = FeatureFilter('device_type', 'desktop')
+        tablet = FeatureFilter('device_type', 'tablet')
+        kpi = KPI('revenue')
+
+        test_revenue_overall = StatisticalTest(kpi, [], variants)
+        test_revenue_mobile = StatisticalTest(kpi, [mobile], variants)
+        test_revenue_desktop = StatisticalTest(kpi, [desktop], variants)
+        test_revenue_tablet = StatisticalTest(kpi, [tablet], variants)
+
+        tests = [test_revenue_overall, test_revenue_mobile, test_revenue_desktop, test_revenue_tablet]
+        multi_test_suite = StatisticalTestSuite(tests, "bh")
+
+        self.assertEqual(multi_test_suite.size, 4)
+        self.assertEqual(multi_test_suite.correction_method, "bh")
+
+    def test_make_derived_kpi(self):
+        numerator = "normal_same"
+        denominator = "normal_shifted"
+        derived_kpi_name = "derived_kpi_one"
+        DerivedKPI(derived_kpi_name, numerator, denominator).make_derived_kpi(self.data)
```

Comment:
`make_derived_kpi()` should be ensured to perform its deeds only once, either here or within the `DerivedKPI.make_derived_kpi()` method

Reply:
good point. I will fix it.